### PR TITLE
CRM-20384, Added Contribution ID column and Filter to Book Keeping report

### PR DESCRIPTION
**Pull request by pradpnayak**
Added option to display and Filter by Financial Account Owner

----------------------------------------
* CRM-20384: bookkeeping transaction report
  https://issues.civicrm.org/jira/browse/CRM-20384

### CODE REVIEW

**Comment by pradpnayak** [comment]
This PR has code to add
1. Contribution ID as Display column and Filter
2. Add Financial Account Owner for Credit and Debit Account

And some code cleanup.

**Comment by eileenmcnaughton** [comment]
when I look at this report it feels non-standard & complex & the reason is that although the civicrm_financial_account is joined on as 2 different aliases for 2 different meanings (debit & credit) they are both defined in the _construct array within one 'table' - I think splitting it into 2 sections in the array 

'debit_financial_account' => array('fields....

'credit_financial_account' => array('fields....

would allow the report to be simpler.

**Comment by JoeMurray** [comment]
Sigh, more nice to have, time consuming refactoring. @pradpnayak can you provide estimate for the suggested refactoring. @colemanw note that this kind of work is what we agreed would not be required for IIDA.

**Comment by colemanw** [comment]
@civicrm-builder test this plese

**Comment by JoeMurray** [comment]
@civicrm-builder test this please

**Comment by colemanw** [comment]
@civicrm-builder test this please

**Comment by colemanw** [comment]
@pradpnayak the test failure looks to be related:

    PHP Fatal error:  Call to undefined method CRM_Financial_BAO_FinancialAccount::getOrganizationNames()
    in /home/jenkins/buildkit/build/core-10112-8oc8y/sites/all/modules/civicrm/CRM/Report/Form/Contribute/Bookkeeping.php on line 217

**Comment by pradpnayak** [comment]
@colemanw i have pushed the missing function.

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -455,4 +455,23 @@ public static function getAllDeferredFinancialAccount() {
...
+      'return' => array("contact_id.organization_name", "contact_id"),
+      'contact_id.is_deleted' => 0,
+      'options' => array('limit' => 0),
+    ));
+    $organizationNames = array();
+    foreach ($result['values'] as $values) {
+      $organizationNames[$values['contact_id']] = $values['contact_id.organization_name'];
+    }
+    return $organizationNames;
+  }
```
Comment (colemanw) [inline CRM/Financial/BAO/FinancialAccount.php]: I think this function ought to take a `$checkPermissions` param which defaults to `TRUE`, and passes that into the api call.

**Comment by monishdeb** [comment]
@JoeMurray for the refactoring, its hard to estimate looking at the underlying Report query, as after dividing the fields and filters into debit and card sections respectively ( under ```$form->_columns```), we need to handle the underlying functions that build SELECT FROM and WHERE and GROUP BY, accordingly. My rough estimate would be 24-26 hr

File: CRM/Financial/BAO/FinancialAccount.php (hunk 1 above)
Comment (monishdeb) [inline CRM/Financial/BAO/FinancialAccount.php]: @colemanw I have made the changes you mentioned

**Pull request merged**